core/src/lib.rs

```diff
@@ -39,3 +48,25 @@ pub mod server;
 /// Shared code for JSON-RPC clients.
 #[cfg(feature = "client")]
 pub mod client;
+
+pub use async_trait::async_trait;
+pub use error::Error;
+
+/// JSON-RPC result.
+pub type RpcResult<T> = std::result::Result<T, Error>;
+
+/// Re-exports for proc-macro library to not require any additional
+/// dependencies to be explicitly added on the client side.
+#[doc(hidden)]
+pub mod __reexports {
+	pub use async_trait::async_trait;
+	pub use serde;
+	pub use serde_json;
+}
+
+pub use beef::Cow;
+pub use serde::{de::DeserializeOwned, Serialize};
```

Comment:
maybe we should re-export `Deserialize` here too..

Reply:
Mayhaps, we are also super inconsistent about reexports, sometimes we put them in `__reexports` module, and sometimes not, I'd leave that for a future PR.